### PR TITLE
Display Symbol loading status in the module list

### PR DESCRIPTION
**Pull request by antonrohr**
4 Commits to enable symbol loading status in the module list. Split up into 4 for ease of review. 

(hidden behind --devmode)

the struct `SymbolLoadingState` follows what `PresetLoadState` is doing, but relies on `enum` (without class) and implicit conversion. If someone has an idea how to make this better, I am all ear.

### CODE REVIEW

**Comment by florian-kuebler** [comment]
Could you provide a screenshot?

File: src/DataViews/include/DataViews/SymbolLoadingState.h [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    }
+    ORBIT_UNREACHABLE();
+  }
+
+  // Returns false for the default color and true for any other color, in the latter case the
+  // parameters are changed.
+  bool GetDisplayColor(unsigned char& red, unsigned char& green, unsigned char& blue) const {
+    switch (state) {
+      case kUnknown:
+        ABSL_FALLTHROUGH_INTENDED;
```
Comment (florian-kuebler) [inline src/DataViews/include/DataViews/SymbolLoadingState.h:44]: aren't we using the std version? `[[fallthrough]];` Also, IIRC, we don't add fallthroughs on empty cases.

File: src/DataViews/include/DataViews/ModulesDataView.h [hunk 2]
```diff
@@ -21,7 +23,8 @@ namespace orbit_data_views {
 class ModulesDataView : public DataView {
  public:
   explicit ModulesDataView(AppInterface* app,
-                           orbit_metrics_uploader::MetricsUploader* metrics_uploader);
+                           orbit_metrics_uploader::MetricsUploader* metrics_uploader,
+                           bool new_ui = absl::GetFlag(FLAGS_devmode));
```
Comment (florian-kuebler) [inline src/DataViews/include/DataViews/ModulesDataView.h:27]: Nit: Not sure if AppInterface has this, but there is App::IsDevMode.

**Review by florian-kuebler — APPROVED** [review]
Nice, only a couple of minor comments. Thanks

File: src/DataViews/include/DataViews/SymbolLoadingState.h (hunk 1 above)
Comment (antonrohr) [reply src/DataViews/include/DataViews/SymbolLoadingState.h]: I removed the explicit fallthrough. Now I dont have to decide between the absl and std version anymore ;) 

Just fyi, we use [[fallthrough]] 13 times in the repo, ABSL_FALLTHROUGH 28 times

File: src/DataViews/include/DataViews/SymbolLoadingState.h [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+#include <absl/base/attributes.h>
+
+#include <string>
+
+#include "OrbitBase/Logging.h"
+
+namespace orbit_data_views {
+
+struct SymbolLoadingState {
```
[the remark this answers is not in the document]
Comment (antonrohr) [reply src/DataViews/include/DataViews/SymbolLoadingState.h:18]: done

File: src/DataViews/include/DataViews/ModulesDataView.h (hunk 2 above)
Comment (antonrohr) [reply src/DataViews/include/DataViews/ModulesDataView.h]: AppInterface doesn't have `IsDevMode`, but ModulesDataView is instantiated from OrbitApp where it is available. I removed the default parameter and used `IsDevMode`

**Comment by antonrohr** [comment]
Regarding Screenshots:
Without devmode, nothing changes. 
With devmode it only shows "Downloading...", "Loading..." and "Loaded" for now. While taking screenshots, I realized the word Downloading does not fully fit in the column. Fixing that now

**Pull request merged**